fix(reacting-to-input-with-state) (#683)
reactjs/pt-br.react.dev: src/content/learn/reacting-to-input-with-state.md

</Sandpack>

Embora esse código seja mais longo do que o exemplo imperativo original, ele é muito menos frágil. Expressar todas as interações como alterações de `state` permite que você introduza posteriormente novos estados visuais sem quebrar os existentes. Também permite que você altere o que deve ser exibido em cada `state` sem alterar a lógica da própria interação.

<Recap>

> * Programação declarativa significa descrever a IU para cada estado visual em vez de microgerenciá-la (imperativa).

Reviewer: ```suggestion
* Programação declarativa significa descrever a UI para cada estado visual em vez de microgerenciá-la (imperativa).
```

> * Programação declarativa significa descrever a IU para cada estado visual em vez de microgerenciá-la (imperativa).

Reviewer: ```suggestion
* Programação declarativa significa descrever a UI para cada estado visual em vez de microgerenciá-la (imperativa).
```

<Solution>

Esse componente tem dois estados visuais: quando a imagem está ativa e quando a imagem está inativa:

* Quando a imagem está ativa, as classes CSS são `background` e `picture picture--active`.
* Quando a imagem está inativa, as classes CSS são `background background--active` e `picture`.

> Uma única variável de `state` booleana é suficiente para lembrar se a imagem está ativa. A tarefa original era remover ou adicionar classes CSS. No entanto, em React, você precisa *descrever* o que deseja ver em vez de *manipular* os elementos da IU. Portanto, você precisa calcular as duas classes CSS com base no estado atual. Você também precisa [interromper a propagação](/learn/responding-to-events#stopping-propagation) para que o clique na imagem não seja registrado como um clique no plano de fundo.

Reviewer: ```suggestion
Uma única variável de `state` booleana é suficiente para lembrar se a imagem está ativa. A tarefa original era remover ou adicionar classes CSS. No entanto, em React, você precisa *descrever* o que deseja ver em vez de *manipular* os elementos da UI. Portanto, você precisa calcular as duas classes CSS com base no estado atual. Você também precisa [interromper a propagação](/learn/responding-to-events#stopping-propagation) para que o clique na imagem não seja registrado como um clique no plano de fundo.
```

</Sandpack>

> Como alternativa, você poderia retornar dois blocos separados de JSX:

Reviewer: ```suggestion
Alternativamente, você poderia retornar dois blocos separados de JSX:
```

</Sandpack>

> Lembre-se de que, se dois blocos JSX diferentes descreverem a mesma árvore, o aninhamento deles (primeiro `<div>` → primeiro `<img>`) deverá estar alinhado. Caso contrário, a ativação de `isActive` recriaria toda a árvore abaixo e [redefiniria seu estado](/learn/preserving-and-resetting-state). É por isso que, se uma árvore JSX semelhante for retornada em ambos os casos, é melhor escrevê-la como uma única parte do JSX.

Reviewer: ```suggestion
Lembre-se de que, se dois blocos JSX diferentes descreverem a mesma árvore, os aninhamentos deles (primeira `<div>` → primeira `<img>`) devem ser correspondentes. Caso contrário, a ativação de `isActive` recriaria toda a árvore abaixo e [redefiniria seu estado](/learn/preserving-and-resetting-state). É por isso que, se uma árvore JSX semelhante for retornada em ambos os casos, é melhor escrevê-la como uma única JSX.
```

Reviewer: Sobre JSX ser substantivo feminino: https://github.com/reactjs/pt-br.react.dev/pull/674#pullrequestreview-1407604221
